### DISCUSSION TITLE
Proxy RTSP stream with cache between client and server

### DISCUSSION THREAD

**Discussion by tirankas** [bluenviron/mediamtx#1030] [Questions]
Hello,
My use case is to have a cached stream during proxying.
I have embedded devices hosting an RTSP server.
Flow is like this:
NX Witness client <-- rtsp-simple-server-server <-- rtsp-simple-client <-- Embedded device rtsp server

The simple server will proxy the data from these devices (simple server client reads from this server and have a server that will publish the contents being proxied).
The main problem is the embedded devices are wirelessly connected to network and latency/Jitter is not guaranteed.
Is there an option in the simple server to cache some amount of time in the client (connecting to the embedded server) before actually sending data via its server?
I cannot touch the top level client that connects to the simple server, its a 3rd party with no option for cache.

Thanks

**Comment by aler9** [comment]
Hello, in RTSP caching can't be introduced at the server level since it's a real-time protocol. What the server can do is waiting for missing packets and re-order packets (and it does starting from v0.19.3), and this helps a lot in case of unstable connections.

Caching must be introduced at the client level, that must cache a specific amount of video playback (e.g. 200ms) and start playback only after the cache is full. This feature is implemented by all major clients.

**Reply by CheSauRie** [reply]
Does ffmpeg support waiting for missing packets and re-order packets, if so what are those flags?
